@here/harp-mapview/lib/Tile.ts

```diff
@@ -1285,6 +1285,35 @@ export class Tile implements CachedResource {
                     materials[techniqueIndex] = material;
                 }
 
+                // Modify the standard textured shader to support height-based coloring.
+                if (
+                    isStandardTexturedTechnique(technique) &&
+                    technique.heightBasedColor === true &&
+                    technique.displacementMap !== undefined
+                ) {
+                    (material as any).onBeforeCompile = (shader: THREE.Shader) => {
+                        shader.fragmentShader = shader.fragmentShader.replace(
+                            "#include <map_pars_fragment>",
+                            `#include <map_pars_fragment>
+    uniform sampler2D displacementMap;
+    uniform float displacementScale;
+    uniform float displacementBias;`
+                        );
+
+                        shader.fragmentShader = shader.fragmentShader.replace(
+                            "#include <map_fragment>",
+                            `#ifdef USE_MAP
+    float disp = texture2D( displacementMap, vUv ).x / ${EarthConstants.MAX_HEIGHT.toFixed(
```

Comment:
I think it would be nicer if we would specify the max( and min) range in the technique instead of using earth max height, b/c this really depends on the source of the displacement map

Reply:
I don't necessarily disagree, but I don't know if we should do this at this moment in the project (specially since the ``TerrainDataSource`` is a really special kind of ``DataSource``).